sharding/contracts/sharding_manager.sol

```diff
@@ -1,253 +1,326 @@
 pragma solidity ^0.4.19;
 
 contract SMC {
-  event TxToShard(address indexed to, int indexed shardId, int receiptId);
-  event CollationAdded(int indexed shardId, uint expectedPeriodNumber, 
-                     bytes32 periodStartPrevHash, bytes32 parentHash,
-                     bytes32 transactionRoot, address coinbase,
-                     bytes32 stateRoot, bytes32 receiptRoot,
-                     int number, bool isNewHead, int score);
-  event Deposit(address collator, int index);
-  event Withdraw(int index);
-
-  struct Collator {
-    // Amount of wei the collator holds
-    uint deposit;
-    // The collator's address
-    address addr;
+  event HeaderAdded(uint indexed shard_id, bytes32 parent_hash,
+                    bytes32 chunk_root, int128 period, int128 height,
+                    address proposer_address, uint proposer_bid,
+                    bytes proposer_signature);
+  event NotaryRegistered(address notary, uint pool_index);
+  event NotaryDeregistered(address notary, uint pool_index, uint deregistered_period);
+  event NotaryReleased(address notary, uint pool_index);
+  event ProposerRegistered(uint pool_index);
+  event ProposerDeregistered(uint index);
+  event ProposerReleased(uint index);
+
+  // Entry in notary_registry
+  struct Notary {
+    // When notary asked for unregistration
+    uint deregistered;
+    // The notary's pool index
+    uint pool_index;
+    // False if notary has not deposited, true otherwise
+    bool deposited;
   }
-
-  struct CollationHeader {
-    bytes32 parentHash;
-    int score;
+  
+  // Entry in proposer_registry
+  struct Proposer {
+    // Shard id of the deposit
+    uint shardId;
+    // Deposit in ether in each shard
+    uint[] deposit;
   }
 
-  struct Receipt {
-    int shardId;
-    uint txStartgas;
-    uint txGasprice;
-    uint value;
-    bytes32 data;
-    address sender;
-    address to;
+  struct CollationHeader {
+    uint256 shard_id;         // pointer to shard
+    bytes32 parent_hash;      // pointer to parent header
+    bytes32 chunk_root;       // pointer to collation body
+    int128 period;            // Period which header should be included
+    int128 height;            // Collation's height
+    address proposer_address; // Proposer's address 
+    uint256 proposer_bid;     // Proposer's bid
+    bytes proposer_signature; // Proposer's signature
   }
 
   // Packed variables to be used in addHeader
   struct HeaderVars {
     bytes32 entireHeaderHash;
     int score;
-    address collatorAddr;
+    address notaryAddr;
     bool isNewHead;
   }
 
-  // collatorId => Collators
-  mapping (int => Collator) public collators;
+  // notaryId => Notaries
+  // mapping (int => Notary) public notaries;
+  address[] public notary_pool;
+  // proposerId => Proposer
+  // mapping (int => Proposer) public proposers;
+  Proposer[] public proposer_pool;
+
+  // Notary registry (deregistered is 0 for not yet deregistered notaries)
```

Comment:
What does this mean "deregistered is 0"? If you are saying that an address will return 0 because the notary is not in the mapping then you can omit this comment because that is common sense.